download-basket/src/main/java/downloadbasket/helpers/DownloadServices.java

```diff
@@ -0,0 +1,203 @@
+
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.util.PropertyUtil;
+import fi.nls.oskari.util.IOHelper;
+import downloadbasket.data.ErrorReportDetails;
+import downloadbasket.data.LoadZipDetails;
+import org.apache.commons.mail.MultiPartEmail;
+
+import javax.activation.DataHandler;
+import javax.activation.DataSource;
+import javax.activation.FileDataSource;
+import javax.mail.internet.MimeBodyPart;
+import javax.mail.internet.MimeMultipart;
+import javax.mail.util.ByteArrayDataSource;
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.UUID;
+import java.util.Locale;
+import java.util.zip.ZipException;
+import java.util.zip.ZipFile;
+import org.springframework.context.MessageSource;
+import fi.nls.oskari.spring.SpringContextHolder;
+
+/**
+ * Download services.
+ *
+ */
+public class DownloadServices {
+	private final Logger LOGGER = LogFactory.getLogger(DownloadServices.class);
+
+	private MessageSource messages;
+
+	/**
+	 * Default Constructor.
+	 */
+	public DownloadServices() {
+	}
+
+	/**
+	 * Load shape-ZIP from Geoserver.
+	 * 
+	 * @param ldz
+	 *            load zip details
+	 * @return filename file name
+	 * @throws IOException
+	 * 
+	 *             Normal way download uses BBOX as the cropping method.
+	 *             Otherwise, filter plugin is used.
+	 * 
+	 */
+	public String loadZip(LoadZipDetails ldz, Locale locale) throws IOException {
+		String realFileName = "";
+		String returnFileName = "";
+		HttpURLConnection conn = null;
+
+		try {
+			LOGGER.debug("WFS URL: " + ldz.getWFSUrl());
+
+			if (ldz.isDownloadNormalWay()) {
+				LOGGER.debug("Download normal way");
+			}
+
+			LOGGER.debug("WFS URL: " + ldz.getWFSUrl());
+			LOGGER.debug("-- filter: " + ldz.getGetFeatureInfoRequest());
+			final URL url = new URL(ldz.getWFSUrl() + ldz.getGetFeatureInfoRequest());
+
+			conn = (HttpURLConnection) url.openConnection();
+			conn.setRequestMethod("GET");
+			conn.setConnectTimeout(600000);
+
+			conn.connect();
+
+			String filename = UUID.randomUUID().toString();
+			String strTempDir = ldz.getTemporaryDirectory();
+			File dir0 = new File(strTempDir);
+			dir0.mkdirs();
+
+			try (InputStream istream = conn.getInputStream();
+					OutputStream ostream = new FileOutputStream(new File(dir0, filename + ".zip"))) {
+				IOHelper.copy(istream, ostream);
+			}
+
+			if (!isValid(new File(dir0, filename + ".zip"))) {
+				ErrorReportDetails erd = new ErrorReportDetails();
+				erd.setErrorFileLocation(realFileName);
+				erd.setWfsUrl(ldz.getWFSUrl());
+				erd.setXmlRequest(ldz.getGetFeatureInfoRequest());
+				erd.setUserEmail(ldz.getUserEmail());
+				erd.setLanguage(locale.getLanguage());
+				sendErrorReportToEmail(erd);
+			}
+
+		} catch (Exception ex) {
+			LOGGER.error("Error: ", ex);
+		}
+		return returnFileName;
+	}
+
+	/**
+	 * Check at is zipfile valid.
+	 * 
+	 * @param file
+	 *            zip file
+	 * @return
+	 */
+	static boolean isValid(final File file) {
```

Comment:
Using try-with-resources would make this function a lot more readable. Also since ZipException extends IOException it doesn't need to be catched separately (since the logic of what-to-do is the same for both exceptions). For example:
```
public static boolean isZipFile(File file) {
    try (ZipFile zipFile = new ZipFile(file)) {
        return true;
    } catch (IOException e) {
        // Maybe log.debug() the exception for good measure
        return false;
    }
}
```